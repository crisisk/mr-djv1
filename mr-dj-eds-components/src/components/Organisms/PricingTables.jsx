import React from 'react';
import Button from '../Atoms/Buttons.jsx';
import { trackEvent } from '../../lib/analytics.js';

const packages = [
  {
    name: 'Brons',
    subtitle: 'Entry-level pakket',
    price: '€495',
    duration: '4 uur showtijd',
    category: ['Private', 'Bedrijfsfeest'],
    features: [
      '4 uur DJ-set',
      'Basis licht- en geluidsset',
      'Persoonlijk intakegesprek',
      'Muziekvoorkeuren formulier',
    ],
    upsells: [
      { id: 'rookmachine', label: 'Rookmachine & sfeerlicht', price: '€75' },
      { id: 'ceremonie', label: 'Ceremonie audio set', price: '€195' },
    ],
    isFeatured: false,
    buttonText: 'Meer Info',
  },
  {
    name: 'Zilver',
    subtitle: 'Meest gekozen',
    price: '€795',
    duration: '5 uur showtijd',
    category: ['Bruiloft'],
    features: [
      '6 uur DJ-set',
      'Uitgebreide lichtshow',
      'DJ + Saxofonist optie',
      '100% dansgarantie',
      'Onbeperkt aantal gasten',
    ],
    upsells: [
      { id: 'sparkular', label: 'Sparkular show', price: '€245' },
      { id: 'sax', label: 'Live saxofonist', price: '€325' },
      { id: 'photobooth', label: '360° photobooth', price: '€425' },
    ],
    isFeatured: true,
    buttonText: 'Boek Nu',
  },
  {
    name: 'Goud',
    subtitle: 'Premium All-Inclusive',
    price: '€1.295',
    duration: '6 uur showtijd + productie',
    category: ['Bruiloft', 'Bedrijfsfeest'],
    features: [
      '8 uur DJ-set',
      'Luxe licht- en geluidsset',
      'DJ + Saxofonist (inbegrepen)',
      'Ceremonie & receptie muziek',
      'Professionele apparatuur',
    ],
    upsells: [
      { id: 'branding', label: 'Custom booth branding', price: '€395' },
      { id: 'host', label: 'Host & vocalist', price: '€295' },
      { id: 'afterfilm', label: 'Aftermovie & content', price: '€495' },
    ],
    isFeatured: false,
    buttonText: 'Vraag Offerte Aan',
  },
];

const PricingCard = ({ pkg }) => {
  const { name, subtitle, price, duration, category, features, upsells, isFeatured, buttonText } = pkg;

  const cardClasses = isFeatured
    ? 'bg-neutral-dark text-neutral-light shadow-2xl transform scale-105'
    : 'bg-neutral-light text-neutral-dark shadow-lg';

  const headerClasses = isFeatured
    ? 'text-secondary border-b border-secondary/50'
    : 'text-primary border-b border-neutral-gray-100';

  const buttonVariant = isFeatured ? 'secondary' : 'primary';

  const handleClick = () => {
    trackEvent('package_cta_click', {
      package_name: name,
      placement: 'design_system_pricing_tables',
    });
  };

  const handleHover = () => {
    trackEvent('package_view', {
      package_name: name,
      placement: 'design_system_pricing_tables',
    });
  };

  return (
    <div className={`relative flex flex-col p-spacing-xl rounded-lg transition duration-300 ${cardClasses}`}>
      {isFeatured && (
        <div className="absolute top-0 right-0 bg-secondary text-neutral-dark text-font-size-small font-bold px-spacing-md py-spacing-xs rounded-tr-lg rounded-bl-lg">
          Populair
        </div>
      )}
      <div className={`pb-spacing-md mb-spacing-md ${headerClasses}`}>
        <h3 className="text-font-size-h3 font-bold">{name}</h3>
        <p className="text-font-size-small opacity-80">{subtitle}</p>
      </div>
      <div className="flex flex-col mb-spacing-lg">
        <span className="text-font-size-h1 font-extrabold">{price}</span>
        {duration && <span className="text-font-size-small opacity-70">{duration}</span>}
        {category && (
          <span className="text-font-size-small opacity-80">
            Geschikt voor: {Array.isArray(category) ? category.join(', ') : category}
          </span>
        )}
      </div>
      <ul className="flex-grow space-y-spacing-sm mb-spacing-xl">
        {features.map((feature, index) => (
          <li key={index} className="flex items-start text-font-size-body">
            <svg
              className={`w-5 h-5 mr-spacing-sm ${isFeatured ? 'text-secondary' : 'text-primary'}`}
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M5 13l4 4L19 7" />
            </svg>
            {feature}
          </li>
        ))}
      </ul>
      {upsells && upsells.length > 0 && (
        <div className="mb-spacing-lg rounded-lg border border-primary/20 bg-primary/5 p-spacing-md">
          <p className="text-font-size-small font-semibold uppercase tracking-wide text-primary mb-spacing-xs">
            Populaire uitbreidingen
          </p>
          <ul className="space-y-spacing-xs text-font-size-small">
            {upsells.map((addon) => (
              <li key={addon.id} className="flex items-center justify-between">
                <span>{addon.label}</span>
                <span className="font-semibold">{addon.price}</span>
              </li>
            ))}
          </ul>
        </div>
      )}
      <Button
        variant={buttonVariant}
        size="lg"
        className="w-full"
        onClick={handleClick}
        onMouseEnter={handleHover}
      >
        {buttonText}
      </Button>
    </div>
  );
};

const PricingTables = () => {
  return (
    <section className="py-spacing-3xl bg-neutral-gray-100">
      <div className="container mx-auto px-spacing-md">
        <h2 className="text-font-size-h2 text-center text-neutral-dark mb-spacing-2xl font-extrabold">
          Onze Pakketten
        </h2>
        <p className="text-center text-font-size-body text-neutral-gray-600 max-w-2xl mx-auto mb-spacing-2xl">
          Combineer basispakketten met uitbreidingen zoals sparkulars, branding en live-musici. De data-layer events volgen elke
          klik zodat marketingteams kunnen optimaliseren.
        </p>
        <div className="grid grid-cols-1 md:grid-cols-3 gap-spacing-xl items-start">
          {packages.map((pkg) => (
            <PricingCard key={pkg.name} pkg={pkg} />
          ))}
        </div>
      </div>
    </section>
  );
};

export default PricingTables;
