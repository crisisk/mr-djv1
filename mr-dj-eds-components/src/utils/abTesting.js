/**
 * A/B Testing Utility
 *
 * Provides automatic 50/50 traffic splitting with cookie persistence
 * for consistent variant assignment across page visits.
 *
 * Features:
 * - Automatic variant assignment (50/50 split)
 * - Cookie persistence for 30 days
 * - Manual override via URL parameter (?variant=B)
 * - Consistent experience across all pages
 */

/**
 * Get a cookie value by name
 * @param {string} name - Cookie name
 * @returns {string|null} Cookie value or null if not found
 */
export const getCookie = (name) => {
  const nameEQ = name + "=";
  const ca = document.cookie.split(';');

  for (let i = 0; i < ca.length; i++) {
    let c = ca[i];
    while (c.charAt(0) === ' ') c = c.substring(1, c.length);
    if (c.indexOf(nameEQ) === 0) return c.substring(nameEQ.length, c.length);
  }

  return null;
};

/**
 * Set a cookie with name, value, and expiration
 * @param {string} name - Cookie name
 * @param {string} value - Cookie value
 * @param {number} days - Number of days until expiration
 */
export const setCookie = (name, value, days) => {
  let expires = "";

  if (days) {
    const date = new Date();
    date.setTime(date.getTime() + (days * 24 * 60 * 60 * 1000));
    expires = "; expires=" + date.toUTCString();
  }

  document.cookie = name + "=" + (value || "") + expires + "; path=/";
};

/**
 * Get or assign A/B test variant
 *
 * Logic:
 * 1. Check URL parameter (?variant=B) - if present, use it and store in cookie
 * 2. Check cookie - if present, use stored variant
 * 3. Assign random variant (50/50 split) and store in cookie
 *
 * @returns {string} 'A' or 'B'
 */
export const getOrAssignVariant = () => {
  const COOKIE_NAME = 'mr_dj_ab_variant';
  const COOKIE_DAYS = 30;

  // Check URL parameter first (manual override)
  const urlParams = new URLSearchParams(window.location.search);
  const urlVariant = urlParams.get('variant');

  if (urlVariant === 'B' || urlVariant === 'A') {
    // URL parameter takes precedence - store in cookie
    setCookie(COOKIE_NAME, urlVariant, COOKIE_DAYS);
    return urlVariant;
  }

  // Check existing cookie
  const cookieVariant = getCookie(COOKIE_NAME);

  if (cookieVariant === 'A' || cookieVariant === 'B') {
    // Valid cookie exists - use it
    return cookieVariant;
  }

  // No valid variant found - assign randomly (50/50 split)
  const randomVariant = Math.random() < 0.5 ? 'A' : 'B';
  setCookie(COOKIE_NAME, randomVariant, COOKIE_DAYS);

  return randomVariant;
};

/**
 * Push variant assignment to GTM dataLayer
 * @param {string} variant - The assigned variant ('A' or 'B')
 */
export const pushVariantToGTM = (variant) => {
  // Initialize dataLayer if it doesn't exist
  window.dataLayer = window.dataLayer || [];

  // Push ab_test_assigned event
  window.dataLayer.push({
    event: 'ab_test_assigned',
    variant: variant,
    ab_test_name: 'contact_form_optimization',
    timestamp: new Date().toISOString()
  });

  console.log(`A/B Test Variant Assigned: ${variant} (pushed to GTM dataLayer)`);
};
