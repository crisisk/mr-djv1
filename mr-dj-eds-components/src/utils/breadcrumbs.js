const HOME_CRUMB = { name: 'Home', url: '/' };

const createBreadcrumbTrail = (...crumbs) => [HOME_CRUMB, ...crumbs];

export const createServiceBreadcrumbs = (name, url) =>
  createBreadcrumbTrail({ name: 'Diensten', url: '/#diensten' }, { name, url });

export const createSimpleBreadcrumbs = (name, url) => createBreadcrumbTrail({ name, url });

export const createLocalSeoBreadcrumbs = ({ city, slug, isBruiloft }) => {
  const serviceCrumb = isBruiloft
    ? { name: 'Bruiloft DJ', url: '/bruiloft-dj' }
    : { name: 'Feest DJ', url: '/feest-dj' };

  const pageCrumb = {
    name: isBruiloft ? `Bruiloft DJ in ${city}` : `DJ in ${city}`,
    url: isBruiloft ? `/${slug}` : `/dj-in-${slug}`,
  };

  return createBreadcrumbTrail(serviceCrumb, pageCrumb);
};

export default {
  createServiceBreadcrumbs,
  createSimpleBreadcrumbs,
  createLocalSeoBreadcrumbs,
};
