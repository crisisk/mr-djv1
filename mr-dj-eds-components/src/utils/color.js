/**
 * Color utility helpers for design token manipulation.
 *
 * Ensures we consistently convert hex color values coming from the
 * design token JSON into rgba strings with an alpha channel. This is
 * shared by multiple organism components so that any formatting fixes
 * automatically propagate.
 */

/**
 * Convert a hex color value to an RGBA string with the provided alpha.
 *
 * @param {string} hex - The hex color (e.g. "#1A2C4B" or "1A2C4B").
 * @param {number} alpha - A number between 0 and 1.
 * @returns {string} RGBA string representation.
 */
export function withAlpha(hex, alpha) {
  if (typeof hex !== 'string' || !hex.trim()) {
    throw new TypeError('Expected a hex color string');
  }

  const normalized = hex.replace('#', '').trim();

  if (!/^[0-9a-fA-F]{6}$/.test(normalized)) {
    throw new Error(`Invalid hex color provided: ${hex}`);
  }

  const bigint = parseInt(normalized, 16);
  const r = (bigint >> 16) & 255;
  const g = (bigint >> 8) & 255;
  const b = bigint & 255;

  return `rgba(${r}, ${g}, ${b}, ${alpha})`;
}

export default {
  withAlpha,
};
