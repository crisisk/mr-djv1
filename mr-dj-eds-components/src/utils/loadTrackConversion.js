let trackConversionModulePromise;

export const loadTrackConversion = () => {
  if (!trackConversionModulePromise) {
    trackConversionModulePromise = import('./trackConversion');
  }

  return trackConversionModulePromise;
};

export default loadTrackConversion;
