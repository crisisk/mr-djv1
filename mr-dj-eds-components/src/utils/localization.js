export const DEFAULT_LOCALE = 'nl';

export const normalizeLocale = (locale) => {
  if (!locale || typeof locale !== 'string') {
    return DEFAULT_LOCALE;
  }

  const lower = locale.toLowerCase();
  const [primary] = lower.split(/[-_]/);
  return primary || DEFAULT_LOCALE;
};

export const resolveLocale = (preferred, fallback = DEFAULT_LOCALE) => {
  const normalized = normalizeLocale(preferred);
  return normalized || fallback;
};

const getTranslationTable = (input) => {
  if (!input) {
    return null;
  }

  if (typeof input === 'string') {
    return { nl: input, en: input };
  }

  if (Array.isArray(input)) {
    return { nl: input, en: input };
  }

  if (typeof input === 'object' && input.translations && typeof input.translations === 'object') {
    return input.translations;
  }

  return null;
};

export const resolveLocalizedValue = (input, locale = DEFAULT_LOCALE) => {
  if (typeof input === 'string') {
    return input;
  }

  const translations = getTranslationTable(input);
  if (!translations) {
    return '';
  }

  const normalized = normalizeLocale(locale);
  if (typeof translations[normalized] === 'string') {
    return translations[normalized];
  }

  if (typeof translations.nl === 'string') {
    return translations.nl;
  }

  const first = Object.values(translations).find((value) => typeof value === 'string');
  return typeof first === 'string' ? first : '';
};

export const resolveLocalizedList = (input, locale = DEFAULT_LOCALE) => {
  if (Array.isArray(input)) {
    return input;
  }

  const translations = getTranslationTable(input);
  if (!translations) {
    return [];
  }

  const normalized = normalizeLocale(locale);
  const candidate = translations[normalized] ?? translations.nl ?? Object.values(translations).find(Array.isArray);
  return Array.isArray(candidate) ? candidate : [];
};

const mergeTranslationTables = (nextValue, existing) => {
  const table = { ...existing };
  const normalizedNext = getTranslationTable(nextValue);

  if (normalizedNext) {
    Object.entries(normalizedNext).forEach(([key, value]) => {
      if (typeof value === 'string' || Array.isArray(value)) {
        table[key] = value;
      }
    });
  }

  if (!table.nl) {
    table.nl = typeof nextValue === 'string' ? nextValue : Array.isArray(nextValue) ? nextValue : '';
  }

  if (table.en === undefined) {
    table.en = table.nl;
  }

  return table;
};

export const createTranslationObject = (value, existing) => ({
  translations: mergeTranslationTables(value, existing?.translations),
});

export const createTranslationList = (value, existing) => ({
  translations: mergeTranslationTables(Array.isArray(value) ? value : [], existing?.translations),
});
