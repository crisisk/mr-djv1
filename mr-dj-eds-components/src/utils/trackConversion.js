/**
 * GA4 Conversion Tracking Utility
 *
 * This module provides functions to track various conversion events for the Mr. DJ website.
 * All events are pushed to the dataLayer for GTM/GA4 processing.
 *
 * GTM Container: GTM-NST23HJX
 * Google Analytics: info@mr-dj.nl
 */

/**
 * Base function to push conversion events to dataLayer
 * @param {Object} eventData - Event data to push
 */
const pushToDataLayer = (eventData) => {
  if (typeof window !== 'undefined' && window.dataLayer) {
    window.dataLayer.push(eventData);
    console.log('GA4 Conversion Tracked:', eventData);
  } else {
    console.warn('dataLayer not found - event not tracked:', eventData);
  }
};

/**
 * Track form submission conversion
 * @param {string} variant - A/B test variant ('A' or 'B')
 * @param {string} eventType - Type of event (e.g., 'bruiloft', 'bedrijfsfeest')
 * @param {string} formType - Type of form ('contact', 'quote', 'availability')
 */
export const trackFormSubmission = (variant = 'A', eventType = '', formType = 'contact') => {
  pushToDataLayer({
    event: 'conversion',
    conversion_type: 'form_submit',
    form_type: formType,
    variant: variant,
    event_type: eventType,
    value: 1,
    currency: 'EUR',
    timestamp: new Date().toISOString()
  });
};

/**
 * Track phone click conversion
 * @param {string} variant - A/B test variant ('A' or 'B')
 * @param {string} location - Where the phone click occurred (e.g., 'header', 'footer', 'contact_page')
 */
export const trackPhoneClick = (variant = 'A', location = 'unknown') => {
  pushToDataLayer({
    event: 'conversion',
    conversion_type: 'phone_click',
    variant: variant,
    click_location: location,
    value: 1,
    currency: 'EUR',
    timestamp: new Date().toISOString()
  });
};

/**
 * Track quote request conversion
 * @param {string} variant - A/B test variant ('A' or 'B')
 * @param {string} eventType - Type of event requesting quote for
 */
export const trackQuoteRequest = (variant = 'A', eventType = '') => {
  pushToDataLayer({
    event: 'conversion',
    conversion_type: 'quote_request',
    variant: variant,
    event_type: eventType,
    value: 1,
    currency: 'EUR',
    timestamp: new Date().toISOString()
  });
};

/**
 * Track availability check conversion
 * @param {string} variant - A/B test variant ('A' or 'B')
 * @param {string} selectedDate - The date selected by user
 */
export const trackAvailabilityCheck = (variant = 'A', selectedDate = '') => {
  pushToDataLayer({
    event: 'conversion',
    conversion_type: 'availability_check',
    variant: variant,
    selected_date: selectedDate,
    value: 1,
    currency: 'EUR',
    timestamp: new Date().toISOString()
  });
};

/**
 * Track pricing table CTA click conversion
 * @param {string} variant - A/B test variant ('A' or 'B')
 * @param {string} packageName - Package name (e.g., 'brons', 'zilver', 'goud')
 * @param {string} packagePrice - Package price
 */
export const trackPricingCTA = (variant = 'A', packageName = '', packagePrice = '') => {
  pushToDataLayer({
    event: 'conversion',
    conversion_type: 'pricing_cta',
    variant: variant,
    package_name: packageName.toLowerCase(),
    package_price: packagePrice,
    value: 1,
    currency: 'EUR',
    timestamp: new Date().toISOString()
  });
};

/**
 * Track navigation to contact page
 * @param {string} variant - A/B test variant ('A' or 'B')
 * @param {string} source - Where the navigation originated from
 */
export const trackContactNavigation = (variant = 'A', source = 'unknown') => {
  pushToDataLayer({
    event: 'conversion',
    conversion_type: 'contact_navigation',
    variant: variant,
    navigation_source: source,
    value: 1,
    currency: 'EUR',
    timestamp: new Date().toISOString()
  });
};

/**
 * Track general button click (for additional tracking needs)
 * @param {string} variant - A/B test variant ('A' or 'B')
 * @param {string} buttonText - Text of the button clicked
 * @param {string} buttonLocation - Location of the button on the page
 */
export const trackButtonClick = (variant = 'A', buttonText = '', buttonLocation = '') => {
  pushToDataLayer({
    event: 'button_click',
    variant: variant,
    button_text: buttonText,
    button_location: buttonLocation,
    timestamp: new Date().toISOString()
  });
};

/**
 * Track WhatsApp chat initiation
 * @param {string} variant - A/B test variant ('A' or 'B')
 */
export const trackWhatsAppClick = (variant = 'A') => {
  pushToDataLayer({
    event: 'conversion',
    conversion_type: 'whatsapp_click',
    variant: variant,
    value: 1,
    currency: 'EUR',
    timestamp: new Date().toISOString()
  });
};

/**
 * Initialize or update user variant for A/B testing
 * @param {string} variant - The variant to set ('A' or 'B')
 */
export const setUserVariant = (variant) => {
  if (typeof window !== 'undefined') {
    // Store variant in sessionStorage for consistency during session
    sessionStorage.setItem('ab_variant', variant);

    // Push to dataLayer
    pushToDataLayer({
      event: 'ab_variant_set',
      variant: variant,
      timestamp: new Date().toISOString()
    });
  }
};

/**
 * Get current user variant
 * @returns {string} Current variant ('A' or 'B')
 */
export const getUserVariant = () => {
  if (typeof window !== 'undefined') {
    return sessionStorage.getItem('ab_variant') || 'A';
  }
  return 'A';
};

// Export all tracking functions as default object
export default {
  trackFormSubmission,
  trackPhoneClick,
  trackQuoteRequest,
  trackAvailabilityCheck,
  trackPricingCTA,
  trackContactNavigation,
  trackButtonClick,
  trackWhatsAppClick,
  setUserVariant,
  getUserVariant
};
