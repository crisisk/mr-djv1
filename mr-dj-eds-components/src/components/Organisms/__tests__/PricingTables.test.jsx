import { beforeEach, describe, expect, it, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/vitest';
import PricingTables from '../PricingTables.jsx';
import { trackPricingCTA, getUserVariant } from '../../utils/trackConversion';

vi.mock('../../utils/trackConversion', () => ({
  trackPricingCTA: vi.fn(),
  getUserVariant: vi.fn(),
}));

describe('PricingTables', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('tracks CTA clicks with user variant and package details', async () => {
    const user = userEvent.setup();

    getUserVariant
      .mockReturnValueOnce('A')
      .mockReturnValueOnce('B')
      .mockReturnValueOnce('B');

    render(<PricingTables />);

    const meerInfoButton = screen.getByRole('button', { name: /meer info/i });
    const boekNuButton = screen.getByRole('button', { name: /boek nu/i });
    const vraagOfferteAanButton = screen.getByRole('button', { name: /vraag offerte aan/i });

    await user.click(meerInfoButton);
    await user.click(boekNuButton);
    await user.click(vraagOfferteAanButton);

    expect(trackPricingCTA).toHaveBeenNthCalledWith(1, 'A', 'Brons', '€495');
    expect(trackPricingCTA).toHaveBeenNthCalledWith(2, 'B', 'Zilver', '€795');
    expect(trackPricingCTA).toHaveBeenNthCalledWith(3, 'B', 'Goud', '€1.295');
  });

  it('matches the snapshot to guard featured styling', () => {
    const { container } = render(<PricingTables />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
