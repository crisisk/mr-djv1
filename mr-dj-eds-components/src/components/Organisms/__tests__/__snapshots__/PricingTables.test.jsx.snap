// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PricingTables matches the snapshot to guard featured styling 1`] = `
<section
  class="py-16 bg-gray-100"
>
  <div
    class="container mx-auto px-4"
  >
    <h2
      class="text-4xl text-center text-[#1A2C4B] mb-12 font-extrabold"
    >
      Onze Pakketten
    </h2>
    <div
      class="grid grid-cols-1 md:grid-cols-3 gap-8 items-center"
    >
      <div
        class="relative flex flex-col p-8 rounded-lg transition duration-300 bg-white text-[#1A2C4B] shadow-lg"
      >
        <div
          class="pb-4 mb-4 text-primary border-b border-gray-100"
        >
          <h3
            class="text-2xl font-bold"
          >
            Brons
          </h3>
          <p
            class="text-sm opacity-80"
          >
            Entry-level pakket
          </p>
        </div>
        <div
          class="flex items-baseline mb-6"
        >
          <span
            class="text-5xl font-extrabold"
          >
            €495
          </span>
          <span
            class="text-base ml-1"
          >
            / event
          </span>
        </div>
        <ul
          class="flex-grow space-y-2 mb-8"
        >
          <li
            class="flex items-start text-base"
          >
            <svg
              class="w-5 h-5 mr-2 text-primary"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 13l4 4L19 7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            4 uur DJ-set
          </li>
          <li
            class="flex items-start text-base"
          >
            <svg
              class="w-5 h-5 mr-2 text-primary"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 13l4 4L19 7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            Basis licht- en geluidsset
          </li>
          <li
            class="flex items-start text-base"
          >
            <svg
              class="w-5 h-5 mr-2 text-primary"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 13l4 4L19 7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            Persoonlijk intakegesprek
          </li>
          <li
            class="flex items-start text-base"
          >
            <svg
              class="w-5 h-5 mr-2 text-primary"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 13l4 4L19 7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            Muziekvoorkeuren formulier
          </li>
        </ul>
        <button
          class="font-semibold rounded-lg transition duration-300 ease-in-out focus:outline-none focus:ring-4 bg-primary-500 text-white hover:bg-primary-600 focus:ring-primary-300 undefined w-full"
          type="button"
        >
          Meer Info
        </button>
      </div>
      <div
        class="relative flex flex-col p-8 rounded-lg transition duration-300 bg-[#1A2C4B] text-white shadow-2xl transform scale-105"
      >
        <div
          class="absolute top-0 right-0 bg-secondary text-[#1A2C4B] text-sm font-bold px-4 py-1 rounded-tr-lg rounded-bl-lg"
        >
          Populair
        </div>
        <div
          class="pb-4 mb-4 text-secondary border-b border-secondary/50"
        >
          <h3
            class="text-2xl font-bold"
          >
            Zilver
          </h3>
          <p
            class="text-sm opacity-80"
          >
            Meest gekozen
          </p>
        </div>
        <div
          class="flex items-baseline mb-6"
        >
          <span
            class="text-5xl font-extrabold"
          >
            €795
          </span>
          <span
            class="text-base ml-1"
          >
            / event
          </span>
        </div>
        <ul
          class="flex-grow space-y-2 mb-8"
        >
          <li
            class="flex items-start text-base"
          >
            <svg
              class="w-5 h-5 mr-2 text-secondary"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 13l4 4L19 7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            6 uur DJ-set
          </li>
          <li
            class="flex items-start text-base"
          >
            <svg
              class="w-5 h-5 mr-2 text-secondary"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 13l4 4L19 7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            Uitgebreide lichtshow
          </li>
          <li
            class="flex items-start text-base"
          >
            <svg
              class="w-5 h-5 mr-2 text-secondary"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 13l4 4L19 7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            DJ + Saxofonist optie
          </li>
          <li
            class="flex items-start text-base"
          >
            <svg
              class="w-5 h-5 mr-2 text-secondary"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 13l4 4L19 7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            100% dansgarantie
          </li>
          <li
            class="flex items-start text-base"
          >
            <svg
              class="w-5 h-5 mr-2 text-secondary"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 13l4 4L19 7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            Onbeperkt aantal gasten
          </li>
        </ul>
        <button
          class="font-semibold rounded-lg transition duration-300 ease-in-out focus:outline-none focus:ring-4 bg-neutral-200 text-neutral-800 hover:bg-neutral-300 focus:ring-neutral-400 undefined w-full"
          type="button"
        >
          Boek Nu
        </button>
      </div>
      <div
        class="relative flex flex-col p-8 rounded-lg transition duration-300 bg-white text-[#1A2C4B] shadow-lg"
      >
        <div
          class="pb-4 mb-4 text-primary border-b border-gray-100"
        >
          <h3
            class="text-2xl font-bold"
          >
            Goud
          </h3>
          <p
            class="text-sm opacity-80"
          >
            Premium All-Inclusive
          </p>
        </div>
        <div
          class="flex items-baseline mb-6"
        >
          <span
            class="text-5xl font-extrabold"
          >
            €1.295
          </span>
          <span
            class="text-base ml-1"
          >
            / event
          </span>
        </div>
        <ul
          class="flex-grow space-y-2 mb-8"
        >
          <li
            class="flex items-start text-base"
          >
            <svg
              class="w-5 h-5 mr-2 text-primary"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 13l4 4L19 7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            8 uur DJ-set
          </li>
          <li
            class="flex items-start text-base"
          >
            <svg
              class="w-5 h-5 mr-2 text-primary"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 13l4 4L19 7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            Luxe licht- en geluidsset
          </li>
          <li
            class="flex items-start text-base"
          >
            <svg
              class="w-5 h-5 mr-2 text-primary"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 13l4 4L19 7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            DJ + Saxofonist (inbegrepen)
          </li>
          <li
            class="flex items-start text-base"
          >
            <svg
              class="w-5 h-5 mr-2 text-primary"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 13l4 4L19 7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            Ceremonie & receptie muziek
          </li>
          <li
            class="flex items-start text-base"
          >
            <svg
              class="w-5 h-5 mr-2 text-primary"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 13l4 4L19 7"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            Professionele apparatuur
          </li>
        </ul>
        <button
          class="font-semibold rounded-lg transition duration-300 ease-in-out focus:outline-none focus:ring-4 bg-primary-500 text-white hover:bg-primary-600 focus:ring-primary-300 undefined w-full"
          type="button"
        >
          Vraag Offerte Aan
        </button>
      </div>
    </div>
  </div>
</section>
`;
